catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/SecureHadoopCatalogOperations.java

```diff
@@ -247,6 +265,20 @@ public void testConnection(
     hadoopCatalogOperations.testConnection(catalogIdent, type, provider, comment, properties);
   }
 
+  @Override
+  public Map<String, CredentialContext> getCredentialContexts(
+      NameIdentifier nameIdentifier, CredentialPrivilege privilege) {
+    if (nameIdentifier.equals(NameIdentifierUtil.getCatalogIdentifier(nameIdentifier))) {
+      return getCatalogCredentialContexts();
+    }
+    return getFilesetCredentialContexts(nameIdentifier, privilege);
+  }
```

Comment:
Do we support credential in the schema level?

Reply:
No

Reply:
The java client only supports `Catalog` and `Fileset`, but there is no extra check for the metadata object type in server side, I'll add it.

Reply:
I mean why don't we support credential in the schema level?

Reply:
The catalog credentials is mainly used for static credentials like AKSK, and the fileset credential is used for dymatic credentials like S3 tokens, is there any scenarios about schema credentials?